src/core/ObservableQuery.ts

```diff
@@ -1207,23 +1216,30 @@ Did you mean to call refetch(variables) instead of refetch({ variables })?`,
 
     const {
       pollingInfo,
-      options: { pollInterval },
+      options: { fetchPolicy, pollInterval },
     } = this;
 
-    if (!pollInterval || !this.hasObservers()) {
+    if (!pollInterval || !this.hasObservers() || fetchPolicy === "cache-only") {
+      if (
+        !this.didWarnCacheOnlyPolling &&
+        pollInterval &&
+        fetchPolicy === "cache-only"
+      ) {
+        invariant.warn(
+          "Cannot poll on 'cache-only' query '%s' and as such, polling is disabled. Please use a different fetch policy.",
+          getOperationName(this.query, "(anonymous)")
+        );
+        this.didWarnCacheOnlyPolling = true;
+      }
+
```

Comment:
The `invariant.warn` would be stripped out in `__DEV__`, but they tests would not be.
```suggestion
      if (__DEV__) {
        if (
          !this.didWarnCacheOnlyPolling &&
          pollInterval &&
          fetchPolicy === "cache-only"
        ) {
          invariant.warn(
            "Cannot poll on 'cache-only' query '%s' and as such, polling is disabled. Please use a different fetch policy.",
            getOperationName(this.query, "(anonymous)")
          );
          this.didWarnCacheOnlyPolling = true;
        }
      }

```

Reply:
Good call.